### PR TITLE
Import tests from dpstd library - part 2/16

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Part 2/16
Please see details in part 1/16 - https://github.com/oneapi-src/oneDPL/pull/885
This PR should be merged into main only after https://github.com/oneapi-src/oneDPL/pull/885

### CODE REVIEW

**Comment by MikeDvorskiy** [comment]
> Part 2/16 Please see details in part 1/16 - #885 This PR should be merged into main only after #885

It seems the PR contains part 1 and part 2 together.
Is possible to have just diff between #885 and this one? Other words, I would prefer review just new changes, based on previous PR #885.

**Comment by SergeyKopienko** [comment]
> > Part 2/16 Please see details in part 1/16 - #885 This PR should be merged into main only after #885
> 
> It seems the PR contains part 1 and part 2 together. Is possible to have just diff between #885 and this one? Other words, I would prefer review just new changes, based on previous PR #885.

Yes, it's true.
You may see the diff for this branch only in desktop git UI tools or after part1 will merged into main branch.

**Comment by SergeyKopienko** [comment]
> > Part 2/16 Please see details in part 1/16 - #885 This PR should be merged into main only after #885
> 
> It seems the PR contains part 1 and part 2 together. Is possible to have just diff between #885 and this one? Other words, I would prefer review just new changes, based on previous PR #885.

@MikeDvorskiy the base for this PR changed to dev/skopienko/add_std_tests_from_dpcpp_part_1
So now you will see here only changes from this PR

**Comment by MikeDvorskiy** [comment]
> > > Part 2/16 Please see details in part 1/16 - #885 This PR should be merged into main only after #885
> > 
> > 
> > It seems the PR contains part 1 and part 2 together. Is possible to have just diff between #885 and this one? Other words, I would prefer review just new changes, based on previous PR #885.
> 
> @MikeDvorskiy the base for this PR changed to `dev/skopienko/add_std_tests_from_dpcpp_part_1` So now you will see here only changes from this PR

Great, thanks

File: test/std/containers/sequences/array/array.comparison/greater_g.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    auto ret = kernel_test();
+    EXPECT_TRUE(ret, "");
```
Comment (MikeDvorskiy) [inline test/std/containers/sequences/array/array.comparison/greater_g.pass.cpp:59]: IMHO, an error message should be meaningful, in order to quick understand what and where is wrong.
That comment regards to other similar places in this PR.

**Comment by SergeyKopienko** [comment]
> > Part 2/16
> 
> "Import tests from std library" - I am not sure that name of PR is right. As far as I can see it is not "C++ std library", is it from "Intel LLVM fork" repo?

About in source repository with this tests: `dpstd library which consists of PSTL and some extensions for sycl`

**Review by MikeDvorskiy — APPROVED** [review]
I looked briefly and didn't notice any potential issues with these tests.

**Review by dmitriy-sobolev — APPROVED** [review]
I see no issues either

**Pull request merged**